sql/handler.cc

```diff
@@ -7639,6 +7639,9 @@ int handler::ha_external_lock(THD *thd, int lock_type)
     }
   }
 
+  if (lock_type == F_UNLCK)
+    (void) table->reset_hlindexes();
```

Comment:
may be, rename it to `unlock_hlindexes()` ? it's only used here, and you've move it from the reset to the unlock code path.

Reply:
Fine with me. I've updated PR and rebased on top of 11.7, but apparently you've already merged it to bb-11.7-release.